pkg/maps/ctmap/ctmap.go

```diff
@@ -410,10 +336,110 @@ func doGC4(m *bpf.Map, filter *GCFilter) int {
 	return deleted
 }
 
+func (f *GCFilter) doFiltering(dstIP net.IP, dstPort uint16, nextHdr, flags uint8, entry *CtEntry) (action int) {
+	action = noAction
+
+	// Delete all entries with a lifetime smaller than f timestamp.
+	if f.Type&GCFilterByTime != 0 &&
+		entry.lifetime < f.Time {
+
+		action = deleteEntry
+	}
+
+	// Delete all entries of the given dstIP that are not filtered
+	if f.Type&GCFilterByIDsToKeep != 0 &&
+		dstIP.Equal(f.IP) &&
+		// only delete TUPLE_F_IN connections since the clean up is made
+		// for ingress policies
+		// #### REVIEWER: what about TUPLE_F_RELATED?
+		flags&TUPLE_F_IN != 0 {
+
+		// Check if the src_sec_id of that entry is still allowed
+		// to talk with the destination IP.
+		if filterRuleCtx, ok := f.IDsToKeep[policy.NumericIdentity(entry.src_sec_id)]; !ok {
+			action = deleteEntry
+		} else {
+			l4RuleCtx := policy.L4RuleContext{
+				Port:  dstPort,
+				Proto: nextHdr,
+			}
+
+			if filterRuleCtx.IsL3Only() {
+				// If the rule is L3-only then check if it's allowed by
+				// L4-only rules.
+				if l4OnlyRules, ok := f.IDsToKeep[policy.InvalidIdentity]; ok {
+					wasAdded, ok := l4OnlyRules[l4RuleCtx]
+					if !ok || !wasAdded.L4Installed {
+						action = deleteEntry
+					}
+				}
+			} else {
+				// If the rule is not L3-only then check if it's allowed by
+				// L3-L4 rules.
+				if l7Rule, ok := filterRuleCtx[l4RuleCtx]; ok {
+					if l7Rule.L4Installed && entry.proxy_port != l7Rule.RedirectPort {
+						action = modifyEntry
+						entry.proxy_port = l7Rule.RedirectPort
+					}
+				} else {
+					action = deleteEntry
```

Comment:
What does this case represent? L4-Only?

Reply:
It means, for example, there's a L3-L4 rule installed which allows port 80, but the port in the CT entry is 82.

Reply:
Ah, cool. If this code isn't otherwise reorganised it'd be useful to have a comment like that to give context. Thanks!